File: src/main/frontend/src/services/responseDto.js
class ResponseDto {
    constructor(success, message, data) {
        this.success = success;
        this.message = message;
        this.data = data;
    }

    static fromPromise(promise) {
        return promise
            .then(data => new ResponseDto(true, "", data))
            .catch(error => new ResponseDto(false, error.message, null));
    }

    static fromResponse(response) {
        const { success, message, data } = response;
        return new ResponseDto(success, message, data);
    }
}

module.exports = ResponseDto;

File: src/main/java/com/seohyeon/bookloan/dto/ResponseDto.java
package com.seohyeon.bookloan.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class ResponseDto<D> {
    private boolean success;
    private String message;
    private D data;

    public static <D> ResponseDto<D> ok(D data){
        return new ResponseDto<>(true, null, data);
    }

    public static ResponseDto<?> error(String message) {
        return new ResponseDto<>(false, message, null);
    }
}